pkg/services/authz/zanzana/zanzana.go

```diff
@@ -8,34 +8,30 @@ import (
 )
 
 const (
-	TypeUser string = "user"
-	TypeTeam string = "team"
+	TypeUser      string = "user"
+	TypeTeam      string = "team"
+	TypeFolder    string = "folder"
+	TypeDashboard string = "dashboard"
 )
 
 const (
 	RelationTeamMember string = "member"
 	RelationTeamAdmin  string = "admin"
+	RelationParent     string = "parent"
 )
 
-func NewObject(typ, id string) string {
-	return fmt.Sprintf("%s:%s", typ, id)
+func NewObject(typ, id, relation string) string {
+	obj := fmt.Sprintf("%s:%s", typ, id)
+	if relation != "" {
+		obj = fmt.Sprintf("%s#%s", obj, relation)
+	}
+	return obj
 }
 
-func NewScopedObject(typ, id, scope string) string {
-	return NewObject(typ, fmt.Sprintf("%s-%s", scope, id))
+func NewScopedObject(typ, id, relation, scope string) string {
```

Comment:
Same here. Probably also why we have this function. I could not figure out a good name for it

Reply:
maybe using namespace, if we go full on zanzibar

```go
func NewNamespacedTuple(typ, id, relation, namespace string) string {
	return NewTuple(typ, fmt.Sprintf("%s-%s", namespace, id), relation)
}
```